File: ros_ws/src/my_robot_tutorials/src/number_counter.cpp
#include <ros/ros.h>
#include <std_msgs/Int64.h>
#include <std_srvs/SetBool.h>

int counter = 0;
ros::Publisher publisher;

void callback_subscription(const std_msgs::Int64& msg)
{
    counter += msg.data;
    std_msgs::Int64 new_msg;
    new_msg.data = counter;
    publisher.publish(new_msg);
}

bool handleResetCounter(std_srvs::SetBool::Request &req, std_srvs::SetBool::Response &res)
{
    if(req.data) {
        counter = 0;
        res.message = "Counter has been reset";
    }
    else {
        res.message = "Counter has not been reset";
    }

    res.success = true;
    ROS_INFO("%s", res.message.c_str());

    return true;
}

int main(int argc, char **argv)
{
    ros::init(argc, argv, "number_counter");
    ros::NodeHandle nh;

    publisher = nh.advertise<std_msgs::Int64>("number_count", 10);
    ros::Subscriber subscriber = nh.subscribe("number", 10, callback_subscription);  
    ros::ServiceServer service = nh.advertiseService("reset_number_count", handleResetCounter);  

    ROS_INFO("number_counter node has been created");

    ros::spin();

    return 0;
} 

File: ros_ws/src/my_robot_tutorials/scripts/number_counter.py
#!/usr/bin/env python

import rospy
from std_msgs.msg import Int64
from std_srvs.srv import SetBool


counter = 0
publisher = None


def main(args=None):
    rospy.init_node("number_counter")

    global publisher
    publisher = rospy.Publisher("number_count", Int64, queue_size=10)
    subscriber = rospy.Subscriber("number", Int64, callback_subscription, queue_size=10)
    service = rospy.Service("reset_number_count", SetBool, handle_reset_counter)    

    rospy.loginfo("number_count node has been created")

    rospy.spin()


def callback_subscription(msg):
    global counter
    global publisher

    counter += msg.data    
    
    new_msg = Int64()
    new_msg.data = counter
    publisher.publish(new_msg)


def handle_reset_counter(req):
    
    global counter
    response = SetBool._response_class()
    
    if req.data:
        counter = 0
        response.message = "Counter has been reset"
    else:
        response.message = "Counter has not been reset"

    response.success = True
    rospy.loginfo(str(response.message))

    return response


if __name__ == "__main__":
    main()